batdata/postprocess/cycle_stats.py

```diff
@@ -1,3 +1,37 @@
 """Utility functions for computing properties of certain cycles"""
+import warnings
 
-# TBD (wardlt): Still working on what goes here
+import numpy as np
+import pandas as pd
+
+from batdata.postprocess.base import CycleSummarizer
+
+
+class CycleTimes(CycleSummarizer):
+    """Capture the start time and duration of a cycle
+
+    The start of a cycle is the minimum test time for any point in the raw data.
+
+    The duration of a cycle is the difference between the start of the next cycle and the start of the cycle.
+    If the start time of the next cycle is unavailable, it is the difference between the test time of the
+    last test time in the raw data and the start fo the cycle.
+    """
+
+    column_names = ['cycle_start', 'cycle_duration']
+
+    def _summarize(self, raw_data: pd.DataFrame, cycle_data: pd.DataFrame):
+        # Compute the starts and durations
+        time_summary = raw_data.groupby('cycle_number')['test_time'].agg(
+            cycle_start=min, cycle_duration=lambda x: max(x) - min(x)
+        ).reset_index()  # reset_index makes `cycle_number` a regular column
+
+        # Compute the duration using the start of the next cycle, if known
+        time_summary['next_diff'] = time_summary['cycle_number'].diff(-1).iloc[:-1]
+        if (time_summary['next_diff'].iloc[:-1] != -1).any():
+            warnings.warn('Some cycles are missing from the dataframe. Time durations for those cycles may be too short')
+        has_next_cycle = time_summary.query('next_diff == -1')
+        time_summary.loc[has_next_cycle.index, 'cycle_duration'] = -time_summary['cycle_start'].diff(-1)[has_next_cycle.index]
```

Comment:
Just checking, but is this needed in case there is only one entry for a given cycle number? I am a bit conflicted here, seeing as
1. the code above it should do the job just fine for cycle numbers with at least 2 rows in the raw_data dataframe: duration is max-min, and start time is min
2. if it happens to be the case that a cycle only has one entry, we should raise an error imo and tell the user to go back to the data ingestors, as it is impossible for the information of an entire cycle to be capture in a single timestep. I encountered something similar to this in the Aurora data, and the best course of action in these cases is to incorporate the "floating" cycle into another and renumber. 

What do you think? Or is this part of the code needed because of something else that I'm missing?